### PR TITLE
Allow failures setting nextWorkerNumberToUse

### PR DESCRIPTION
**Pull request by hmitnflx**
Issue:
Sometimes, the underlying store isn't available for the extensive writes made to update nextWorkerNumberToUse. An exception is thrown cascaded upstream preventing further writes and updates to worker from JobActor. To handle it, we tolerate some (consecutive) write failures for this call before propagating the exception upstream.

Invariant:
This is fine from a constraints perspective. Here's why: On new mantis master leader election, job and worker state is read from the DB. If the worker count was synced correctly no issue. If not and there were extra workers than workerMax stored in DB, those extra workers will be killed when TEs send heartbeat. If not and workerMax in DB is higher than actual workers, we'll schedule those workers on leader election.

### Context

Explain context and other details for this pull request.

### Checklist

- [ ] `./gradlew build` compiles code correctly
- [ ] Added new tests where applicable
- [ ] `./gradlew test` passes all tests
- [ ] Extended README or added javadocs where applicable

### CODE REVIEW

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java [hunk 1]
```diff
@@ -1202,21 +1205,28 @@ static class WorkerNumberGenerator {
...
 
         private void advance(MantisJobMetadataImpl mantisJobMetaData, MantisJobStore jobStore) {
+            currLimit += incrementStep;
             try {
-                currLimit += incrementStep;
-
                 mantisJobMetaData.setNextWorkerNumberToUse(currLimit, jobStore);
+                pendingFlushCnt.set(0);
             } catch (Exception e) {
+                if (pendingFlushCnt.incrementAndGet() > MAX_PENDING_FLUSH_CNT) {
```
Comment (Andyz26) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java:1213]: the write failure will still throw the runtime exception?

Comment (hmitnflx) [reply mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java:1213]: oops, missed an early return here during a refactoring I did. Thanks!

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java [hunk 2]
```diff
@@ -1203,28 +1210,47 @@ static class WorkerNumberGenerator {
...
+                LOGGER.error("Exception setting next Worker number to use, " +
+                    "{} consecutive failed attempts", MAX_PENDING_FLUSH_CNT, e);
                 throw new RuntimeException("Unexpected error setting next worker number to use", e);
             }
         }
 
+        private void setNextWorkerNumberWithRetries(MantisJobMetadataImpl mantisJobMetaData, MantisJobStore jobStore, int value) throws Exception {
+            int attempts = 0;
+            Exception exception = null;
+            while (attempts < MAX_ATTEMPTS) {
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java:1228]: can we retry infinitely?

**Review by sundargates — APPROVED** [review]
The change looks good! Ship it!

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java (hunk 2 above)
Comment (hmitnflx) [reply mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java:1226]: I don't prefer an infinite retry for now, we can customize it in the future. For now, I increased it from 3 to 10

**Pull request merged**